src/libraries/System.Formats.Tar/tests/TarEntry/TarEntry.Conversion.Tests.Base.cs

```diff
@@ -0,0 +1,209 @@
+            if (format is TarEntryFormat.Pax)
+            {
+                PaxTarEntry paxEntry = firstEntry as PaxTarEntry;
+                Assert.Contains("atime", paxEntry.ExtendedAttributes);
+                Assert.Contains("ctime", paxEntry.ExtendedAttributes);
+            }
+            else if (format is TarEntryFormat.Gnu)
+            {
+                GnuTarEntry gnuEntry = firstEntry as GnuTarEntry;
+                gnuEntry.AccessTime = TestAccessTime;
+                gnuEntry.ChangeTime = TestChangeTime;
+            }
+
+            return firstEntry;
+        }
+
+        private TarEntry ConvertAndVerifyEntry(TarEntry originalEntry, TarEntryType entryType, TarEntryFormat formatToConvert)
+        {
+            TarEntry convertedEntry = InvokeTarEntryConversionConstructor(formatToConvert, originalEntry);
+
+            CheckConversionType(convertedEntry, formatToConvert);
+            Assert.Equal(formatToConvert, convertedEntry.Format);
+
+            TarEntryType convertedEntryType = GetTarEntryTypeForTarEntryFormat(entryType, formatToConvert);
+            Assert.Equal(convertedEntryType, convertedEntry.EntryType);
+
+            Assert.Equal(originalEntry.Gid, convertedEntry.Gid);
+            Assert.Equal(originalEntry.Uid, convertedEntry.Uid);
+            Assert.Equal(originalEntry.Mode, convertedEntry.Mode);
+            Assert.Equal(originalEntry.ModificationTime, convertedEntry.ModificationTime);
+
+            if (originalEntry.EntryType is TarEntryType.V7RegularFile or TarEntryType.RegularFile)
+            {
+                Assert.Same(originalEntry.DataStream, convertedEntry.DataStream);
+            }
+            else if (originalEntry.EntryType is TarEntryType.SymbolicLink or TarEntryType.HardLink)
+            {
+                Assert.Equal(originalEntry.LinkName, convertedEntry.LinkName);
+            }
+            else if (originalEntry.EntryType is TarEntryType.BlockDevice or TarEntryType.CharacterDevice)
+            {
+                PosixTarEntry originalPosixTarEntry = originalEntry as PosixTarEntry;
+                PosixTarEntry convertedPosixTarEntry = convertedEntry as PosixTarEntry;
+                Assert.Equal(originalPosixTarEntry.DeviceMajor, convertedPosixTarEntry.DeviceMajor);
+                Assert.Equal(originalPosixTarEntry.DeviceMinor, convertedPosixTarEntry.DeviceMinor);
+            }
+
+            if (formatToConvert is TarEntryFormat.Pax &&
+                originalEntry.Format is TarEntryFormat.Pax or TarEntryFormat.Gnu)
+            {
+                GetTimestamps(originalEntry, out DateTimeOffset expectedATime, out DateTimeOffset expectedCTime);
+                PaxTarEntry paxEntry = convertedEntry as PaxTarEntry;
+                Assert.Equal(expectedATime, GetDateTimeOffsetFromTimestampString(paxEntry.ExtendedAttributes, "atime"));
+                Assert.Equal(expectedCTime, GetDateTimeOffsetFromTimestampString(paxEntry.ExtendedAttributes, "ctime"));
+            }
```

Comment:
```suggestion
            }
            else
            {
                // whatever these values would have when converting from V7
            }
```

Reply:
Good point. I improved this code and added all the missing comparisons whe converting not just from V7 but also from Ustar.